schema/query/predicate_validator.go

```diff
@@ -15,6 +15,17 @@ func validateExpressions(exps []Expression, validator schema.Validator) error {
 	return nil
 }
 
+func prepareExpressions(exps []Expression, validator schema.Validator) ([]Expression, error) {
+	for i, exp := range exps {
+		if nexp, err := exp.Prepare(validator); err != nil {
+			return nil, err
+		} else {
+			exps[i] = nexp[0]
```

Comment:
Is this the right way to do it? Does `exps` can be reused, and not defining `newExpressions` variable?

Reply:
actually, `[]Expression` also implements `Expression`... so that should be a valid return value as well, and you don't need the index `0` or `...` in append.